services/director-v2/src/simcore_service_director_v2/core/dynamic_sidecar_settings.py

```diff
@@ -0,0 +1,337 @@
+from models_library.utils.enums import StrAutoEnum
+from pydantic import Field, PositiveFloat, PositiveInt, validator
+from settings_library.base import BaseCustomSettings
+from settings_library.r_clone import RCloneSettings as SettingsLibraryRCloneSettings
+from settings_library.utils_service import DEFAULT_FASTAPI_PORT
+
+from ..constants import DYNAMIC_SIDECAR_DOCKER_IMAGE_RE
+
+_logger = logging.getLogger(__name__)
+
+MINS = 60
+
+
+SERVICE_RUNTIME_SETTINGS: str = "simcore.service.settings"
+SERVICE_REVERSE_PROXY_SETTINGS: str = "simcore.service.reverse-proxy-settings"
+SERVICE_RUNTIME_BOOTSETTINGS: str = "simcore.service.bootsettings"
+
+SUPPORTED_TRAEFIK_LOG_LEVELS: set[str] = {"info", "debug", "warn", "error"}
+
+
+class VFSCacheMode(str, Enum):
+    __slots__ = ()
+
+    OFF = "off"
+    MINIMAL = "minimal"
+    WRITES = "writes"
+    FULL = "full"
+
+
+class EnvoyLogLevel(StrAutoEnum):
+    TRACE = auto()
+    DEBUG = auto()
+    INFO = auto()
+    WARNING = auto()
+    ERROR = auto()
+    CRITICAL = auto()
+
+    def to_log_level(self) -> str:
+        lower_log_level: str = self.value.lower()
+        return lower_log_level
+
+
+class RCloneSettings(SettingsLibraryRCloneSettings):
+    R_CLONE_DIR_CACHE_TIME_SECONDS: PositiveInt = Field(
+        10,
+        description="time to cache directory entries for",
+    )
+    R_CLONE_POLL_INTERVAL_SECONDS: PositiveInt = Field(
+        9,
+        description="time to wait between polling for changes",
+    )
+    R_CLONE_VFS_CACHE_MODE: VFSCacheMode = Field(
+        VFSCacheMode.MINIMAL,
+        description="used primarly for easy testing without requiring requiring code changes",
+    )
+
+    @validator("R_CLONE_POLL_INTERVAL_SECONDS")
+    @classmethod
+    def enforce_r_clone_requirement(cls, v: int, values) -> PositiveInt:
+        dir_cache_time = values["R_CLONE_DIR_CACHE_TIME_SECONDS"]
+        if v >= dir_cache_time:
+            msg = f"R_CLONE_POLL_INTERVAL_SECONDS={v} must be lower than R_CLONE_DIR_CACHE_TIME_SECONDS={dir_cache_time}"
+            raise ValueError(msg)
+        return v
+
+
+class DynamicSidecarProxySettings(BaseCustomSettings):
+    DYNAMIC_SIDECAR_CADDY_VERSION: str = Field(
+        "2.6.4-alpine",
+        description="current version of the Caddy image to be pulled and used from dockerhub",
+    )
+    DYNAMIC_SIDECAR_CADDY_ADMIN_API_PORT: PortInt = Field(
+        default_factory=lambda: random.randint(1025, 65535),  # noqa: S311
+        description="port where to expose the proxy's admin API",
+    )
+
+
+class DynamicSidecarEgressSettings(BaseCustomSettings):
+    DYNAMIC_SIDECAR_ENVOY_IMAGE: DockerGenericTag = Field(
+        "envoyproxy/envoy:v1.25-latest",
+        description="envoy image to use",
+    )
+    DYNAMIC_SIDECAR_ENVOY_LOG_LEVEL: EnvoyLogLevel = Field(
+        default=EnvoyLogLevel.ERROR,  # type: ignore
+        description="log level for envoy proxy service",
+    )
+
+
+class DynamicSidecarSettings(BaseCustomSettings):
```

Comment:
THOUGHT: where are all these settings coming for. Is it exactly the ones of the dyanmic sidecar service? can we perhaps use `service_library` to have common settings? what do you think?

Reply:
I just saw it now, this needs a separate PR to better manage all the settings. It's gonna be very messy. But I will have one out.